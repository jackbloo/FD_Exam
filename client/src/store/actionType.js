export const LOGIN = 'LOGIN'
export const FETCHPASSWORD = 'FETCHPASSWORD'
export const LOGOUT = 'LOGOUT'
export const FETCHDETAIL = 'FETCH DETAIL'
export const SETEDIT = 'SETEDIT'
export const SHOW = 'SHOW'
export const UNSHOW = 'UNSHOW'
export const LOAD = 'LOAD'
export const UNLOAD = 'UNLOAD'
export const DELETE = 'DELETE'