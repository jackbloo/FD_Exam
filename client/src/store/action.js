import {LOGIN, LOGOUT, FETCHPASSWORD,SHOW,UNSHOW,LOAD,UNLOAD, FETCHDETAIL} from '../store/actionType'


export const show = () => ({type:SHOW})
export const unshow = () => ({type:UNSHOW})
export const login = () => ({type:LOGIN})
export const logout = () => ({type:LOGOUT})
export const load = () => ({type:LOAD})
export const unload = () => ({type: UNLOAD})
export const fetchPassword = (data) =>  ({type:FETCHPASSWORD, data})


// export const deleteOnePass = (id) => dispatch => {
//     dispatch({type:LOAD})
//     pm
//     .doc(id)
//     .delete()
//     .then(function() {
//         dispatch({type:UNLOAD})
//         toast.success('Your Password is deleted')
//     }).catch(function(error) {
//         toast.error(error)
//     });
// }
