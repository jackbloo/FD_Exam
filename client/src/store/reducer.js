import {GETDATA} from './actionType'

const initialState = {
    editorChoice:[],
    latestArticles:[],
    latestReview:[],
    navbar: [{id:'skincare'},{id:'make up'},{id:'body'},{id:'hair'},{id:'fragrance'},{id:'nails'},{id:'tools'},{id:'brands'}],
    linkData:[{id: 'About Us'}, {id: 'Terms & Conditions'}, {id: 'Awards'},{id: 'Feedback'},{id: 'Privacy Policy'}, {id: 'Newsletter'}, {id: 'Content'},{id: 'Help'}]
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case GETDATA:
            const data = action.data
            return {
                ...state,
                editorChoice: [...data["editor's choice"]],
                latestArticles:[...data["latest articles"]],
                latestReview:[...data["latest review"]]
            }
        default:
            return state     
    }
 
}

export default reducer