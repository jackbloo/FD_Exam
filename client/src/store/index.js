import { configureStore } from '@reduxjs/toolkit'
import reducer from '../store/reducer'

const store = configureStore({reducer: reducer})

export default store