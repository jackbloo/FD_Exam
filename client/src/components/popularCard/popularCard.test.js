import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import PopularCard from './popularCard'
import { Provider} from 'react-redux'
import store from '../../store'


describe('Unit Testing of PopularCard', () => {
    const data = {
        image:'131231.png',
        title: 'Test',
        author: 'Test-1',
        published_at: '2 hours ago'
    }

    it('Should Render Card match snapshot', ()=>{
 
     render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-card')
        expect(card).toMatchSnapshot()
    })


    it('Should Render Card without crashing', ()=>{
     render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-card')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
        expect(card).toHaveTextContent("Embrace the Curl---May our curls pop and never stop!")
        expect(card).not.toHaveTextContent('Facebook.com')
    })

    
    it('Should Render popular image', ()=>{
     render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-image')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render popular title', ()=>{
     render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-title')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render popular user', ()=>{
     render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-user')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render popular list', ()=>{
     render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-list')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })
    it('Should Render popular comment', ()=>{
        render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-comment')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render popular description', ()=>{
        render(
            <Provider store={store}>
                    <PopularCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('popular-description')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

})

