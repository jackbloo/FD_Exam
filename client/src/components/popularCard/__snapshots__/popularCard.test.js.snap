// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing of PopularCard Should Render Card match snapshot 1`] = `
<div
  class="sc-bdvvtL fUOFOh"
  data-testid="popular-card"
>
  <div
    class="sc-gsDKAQ gjgvoN"
  >
    <img
      class="sc-dkPtRN foWWYS"
      data-testid="popular-image"
      src="/assets/NaN.png"
    />
    <div
      class="sc-hKwDye jiwCiS"
    >
      <div
        class="sc-eCImPb gJyOyQ"
        data-testid="popular-title"
      >
        Embrace the Curl
      </div>
    </div>
    <div
      class="sc-gKclnd beZVdp"
    >
      <div
        class="sc-iCfMLu hQNmeK"
      >
        <div
          class="sc-furwcr bDSHle"
          data-testid="popular-user"
        >
          <i
            class="sc-pVTFL cwLJSS fa fa-user icon"
          />
          -
        </div>
        <div
          class="sc-furwcr bDSHle"
          data-testid="popular-list"
        >
          <i
            class="sc-pVTFL cwLJSS fa fa-list icon"
          />
          -
        </div>
        <div
          class="sc-furwcr bDSHle"
          data-testid="popular-comment"
        >
          <i
            class="sc-pVTFL cwLJSS fa fa-comment icon"
          />
          -
        </div>
      </div>
      <div
        class="sc-jRQBWg SbPbW"
        data-testid="popular-description"
      >
        May our curls pop and never stop!
      </div>
    </div>
  </div>
</div>
`;
